src/pyOpenMS/pyopenms/__init__.py

```
@@ -56,7 +56,82 @@
+
+    _original_resolve = _pa_fs._resolve_filesystem_and_path
+
+    def _patched_resolve(path, filesystem=None, *, memory_map=False):
+        """Patched version that reuses cached LocalFileSystem instances."""
+        # Handle file-like objects
+        if not _pa_fs._is_path_like(path):
+            if filesystem is not None:
+                raise ValueError(
+                    "'filesystem' passed but the specified path is file-like, so"
+                    " there is nothing to open with 'filesystem'."
+                )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)